File: code/d2203.pas
program Bai203;
  var n, a: longint;
  function SNT(x: longint) : boolean;
    var i :longint;
    begin
    if x < 2 then exit(false);
    for i := 2 to x div 2 do 
        if x mod i = 0 then exit(false);
        exit(true);
    end;
begin
  readln(n);
  for a := (n div 2) downto 2 do
      begin
      if (SNT(a) = true) and (SNT(n - a)) = true then
         begin
         writeln (a, '   ', n-a);
         exit;
         end;
      end;
end.

File: code/d2203.cpp
#include <iostream>
using namespace std;

bool snt(int n){
    int x;
    cin>>n;
    if (n<2)
        return false;
    for (x=2; x<=n-1; x++)
        if (n%x==0)
            return false;
    return true;
}

int main() 
{
    int n,a;
    cin>>n;
    for (a=n/2; a>=2; a--)
        if (snt(a) && snt(n-a)){
            cout<<a<<" "<<n-a;
            return 0;
        }
    return 0;
}